tests/integration/cairo1_hint_locals/deploy/deploy_test.cairo

```diff
@@ -63,3 +63,23 @@ fn test_deploy_with_ctor_invalid_calldata() {
     };
     deploy(prepared_contract).unwrap();
 }
+
+#[test]
+fn test_deploy_with_ctor_panic() {
+    let mut constructor_calldata = ArrayTrait::new();
+    constructor_calldata.append(1);
+    constructor_calldata.append(2);
+
+
+    let class_hash = declare('with_ctor_panic').unwrap();
+    assert(class_hash != 0, 'declared class_hash != 0');
+
+    let prepare_result = prepare(class_hash, constructor_calldata).unwrap();
+
+    let prepared_contract = PreparedContract {
+        contract_address: prepare_result.contract_address,
+        class_hash: prepare_result.class_hash,
+        constructor_calldata: prepare_result.constructor_calldata
+    };
+    deploy(prepared_contract).unwrap();
```

Comment:
We should be able to assert error message as well, no only check if it panicked

Reply:
[b5a0f7c](https://github.com/software-mansion/protostar/pull/1716/commits/b5a0f7c6a394c91b25063499a94301c5e5a73480)